src/superpmi-collect/superpmicollect.cs

```diff
@@ -0,0 +1,777 @@
+        {
+            try
+            {
+                File.Delete(filePath);
+            }
+            catch(Exception ex)
+            {
+                string err = string.Format("Error deleting file \"{0}\": {1}", filePath, ex.Message);
+                s_errors += err + System.Environment.NewLine;
+                Console.Error.WriteLine(err);
+            }
+        }
+
+        private static void CreateTempDirectory(string tempPath)
+        {
+            if (tempPath == null)
+            {
+                tempPath = Path.GetTempPath();
+            }
+            s_tempDir = Path.Combine(tempPath, Path.GetRandomFileName() + "SPMI");
+            if (Directory.Exists(s_tempDir))
+            {
+                throw new SpmiException("temporary directory already exists: " + s_tempDir);
+            }
+            DirectoryInfo di = Directory.CreateDirectory(s_tempDir);
+        }
+
+        private static void ChooseFilePaths(string outputMchPath)
+        {
+            s_baseFailMclFile  = Path.Combine(s_tempDir, "basefail.mcl");
+            s_finalFailMclFile = Path.Combine(s_tempDir, "finalfail.mcl");
+            s_baseMchFile      = Path.Combine(s_tempDir, "base.mch");
+            s_cleanMchFile     = Path.Combine(s_tempDir, "clean.mch");
+
+            if (outputMchPath == null)
+            {
+                s_saveFinalMchFile = false;
+                s_finalMchFile = Path.Combine(s_tempDir, "final.mch");
+                s_tocFile = Path.Combine(s_tempDir, "final.mch.mct");
+            }
+            else
+            {
+                s_saveFinalMchFile = true;
+                s_finalMchFile = Path.GetFullPath(outputMchPath);
+                s_tocFile = s_finalMchFile + ".mct";
+            }
+        }
+
+        private static int RunProgram(string program, string arguments)
+        {
+            if (program.LastIndexOf(".py") != -1)
+            {
+                string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                arguments = "-u " + program; 
+                program = "python";
+            }
+
+            else if (program.LastIndexOf(".dll") != -1)
+            {
+               string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+               arguments = program;
+               program = "dotnet";
+            }
+            
+            // If the program is a script, move the program name into the arguments, and run it
+            // under the appropriate shell.
+            if (Global.IsWindows)
+            {
+                if ((program.LastIndexOf(".bat") != -1) || (program.LastIndexOf(".cmd") != -1))
+                {
+                    string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                    arguments = "/c " + program + programArgumentSep + arguments;
+                    program = Environment.GetEnvironmentVariable("ComSpec"); // path to CMD.exe
```

Comment:
What if we are running in a SysWOW64 window? ComSpec is going to be the normal system32 cmd.exe, not the SysWOW64 cmd.exe. I'm not sure if this makes a difference for superpmi, but it's something to be aware of